### PR TITLE
Error improvements for workload readiness

### PR DESCRIPTION
**Pull request by stevejoachim**
## Pull request type

Please check the type of change your PR introduces:
- [x] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/workload_ready_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	c.Assert(ready, DeepEquals, true)
+	c.Assert(err, IsNil)
+}
+
+func (s *WorkloadReadySuite) TestWaitOnDeploymentReady(c *C) {
+	cp := cliParams{"dep", "default", false, 5, 1, 1, 1, 2, "Running"}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second/4)
+	defer cancel()
+	err := WaitOnDeploymentReady(ctx, getCli(cp), cp.namespace, cp.name)
+	c.Assert(err, IsNil)
```
Comment (pavannd1) [inline pkg/kube/workload_ready_test.go:55]: @stevejoachim Looks like this is failing CI
```
Specified 5 replicas and only have 1: Context done while polling: context deadline exceeded ("Specified 5 replicas and only have 1: Context done while polling: context deadline exceeded")
```

Comment (stevejoachim) [reply pkg/kube/workload_ready_test.go]: Thanks for catching that error. I just updated the PR.
In general, the tests were not doing anything useful as written. I was just using them during development to force the different error states.
To make them more useful, we could run a table-driven test that checks the different error messages. For an example, I rewrote `TestWaitOnDeploymentReady` this way. Might be overkill though. What do you think?

Comment (pavannd1) [reply pkg/kube/workload_ready_test.go]: The test looks good. You could maybe add cases for the remaining errors in the function and a case for nil error.

Comment (stevejoachim) [reply pkg/kube/workload_ready_test.go]: Thanks for reviewing. I'll go ahead and add those additional tests.

**Review by ewhamilton — COMMENTED** [review]
Looks good to me. Will let @pavannd1 approve since he reviewed previously.

**Review by pavannd1 — APPROVED** [review]
LGTM

**Pull request merged**